superset-frontend/src/views/components/MenuRight.tsx

```diff
@@ -144,6 +168,18 @@ const RightMenu = ({
     },
   ];
 
+  const hasFileUploadEnabled = () => {
+    SupersetClient.get({
+      endpoint: '/api/v1/database/upload_enabled',
+    }).then(({ json }: Record<string, any>) => {
+      setAllowUploads(json.can_upload);
+    });
+  };
+
+  useEffect(() => {
+    hasFileUploadEnabled();
+  }, []);
```

Comment:
```suggestion
  useEffect(() => hasFileUploadEnabled(), []);
```
Small nit, not blocking, but this can be written on one line.